nifi-framework-bundle/nifi-framework/nifi-framework-cluster-protocol/src/main/java/org/apache/nifi/cluster/protocol/AbstractNodeProtocolSender.java

```diff
@@ -156,6 +158,23 @@ public ClusterWorkloadResponseMessage clusterWorkload(final ClusterWorkloadReque
         throw new ProtocolException("Expected message type '" + MessageType.CLUSTER_WORKLOAD_RESPONSE + "' but found '" + responseMessage.getType() + "'");
     }
 
+    @Override
+    public NodeStatusesResponseMessage nodeStatuses(final NodeStatusesRequestMessage msg) throws ProtocolException {
+        final InetSocketAddress serviceAddress;
+        try {
+            serviceAddress = getServiceAddress();
+        } catch (IOException e) {
+            throw new ProtocolException("Failed to getServiceAddress due to " + e, e);
```

Comment:
Although the `due to` approach with concatenated Exception.toString() still exists in some places, it is unnecessarily repetitive, since the Exception stack trace includes all those details.
```suggestion
            throw new ProtocolException("Failed to get Service Address", e);
```